tests/test_models.py

```diff
@@ -645,6 +645,26 @@ def test_node_assignment():
     darkmodel.meta.exposure = darkexp
 
 
+# Test that a currently undefined attribute can be assigned using dot notation
+# so long as the attribute is defined in the corresponding schema.
+def test_node_new_attribute_assignment():
```

Comment:
This should be in `test_stnode.py` as it is testing adding to the node object.

Reply:
So moved.

Reply:
@WilliamJamieson are you asking for it to work only on `DataModel`objects (instead of node objects), or both? If the former, I can imagine many cases where the actions on node objects is desired. So I assume you mean both. I'll take a look at that.

Reply:
In looking at the code, doesn't it already work for DataModel objects? I did a test interactively and it appeared to work fine. Is the issue that a corresponding test is needed?